### ISSUE TITLE
Possible API inconsistency

### ISSUE THREAD

**Issue by shaomeng** [NCAR/VAPOR#3227]
If I recall correctly, there was an effort to refactor all grids in VAPOR so that they appear to be 3D. For example, a 2D grid in nature would appear as 3D with the Z dimension has a length of 1. Part of the refactoring work was to provide new API calls that return/fill a fixed-length array of 3 to indicate grid dimension, and deprecate old API calls that return/fill a std::vector<> with either 2 or 3 elements in it. 

If my recollection was correct, there seems to be unfinished job. Specifically, the DataMgr::GetDimLensAtLevel() function still provides the old API with std::vector<> and either 2 or 3 elements in it.

**Comment by clyne** [comment]
Correct. The DataMgr::GetDimLensAtLevel() wasn't refactored because so much of the code makes use of the returned vector length to determine the dimensionality of the data.  The  chance of introducing an undetected regression is very high. With the creation of the Python testing framework it might be worth revisiting this issue.
